pkg/apis/agent/v1/klusterletaddonconfig_types.go

```diff
@@ -50,9 +50,6 @@ type KlusterletAddonConfigSpec struct {
 
 	// CertPolicyControllerConfig defines the configurations of CertPolicyController addon agent.
 	CertPolicyControllerConfig KlusterletAddonAgentConfigSpec `json:"certPolicyController"`
-
-	// IAMPolicyControllerConfig defines the configurations of IamPolicyController addon agent.
-	IAMPolicyControllerConfig KlusterletAddonAgentConfigSpec `json:"iamPolicyController"`
```

Comment:
there are also 2 places need to update:
1. the console creates klutereletAddonConfig CR for each cluster, should remove IAMPolicyControllerConfig filed in the CR.
2. should update the CRD in MCH operator repo although we only change the comment to mark it deprecated.

Reply:
Thanks @zhiweiyin318! We didn't consider the deprecation/upgrade requirements.
- We removed it from the console here: https://github.com/stolostron/console/pull/3441
- We have an MCH PR here: https://github.com/stolostron/multiclusterhub-operator/pull/1466/files#diff-bef7823a7fbb24b42ab272467ae82c939bbb8251bfdeb2b20eefd467388be09a
  - For this, it seems we should mark it deprecated an remove the requirement for the field?

Reply:
yes, for the upgrade case, like when the ACM is upgrade from 2.10 to 2.11, the filed IAMPolicyControllerConfig in klutereletAddonConfig CR for each cluster will not be deleted,  right?  if the filed is removed from CRD in 2.11,  may meet an error.
we can mark it deprecated and remove the requirement for the field in this release.  and could consider to remove it from CRD after 2-3 releases when there is no this filed in all CRs.

Reply:
I made a test for the upgrade case. looks like everything is ok and the filed in CR is also deleted automatically . 
kindly ping @qiujian16  need your comment for this case deleting a filed from v1 API?

Reply:
I think we should deperecate this field instead of removing it immediately. My worries is someone might set different configurations in this field before, which will make upgrade fails. And we need to ensure the consumers of this API are aware the deprecation before it is removed.

Reply:
Thanks all!  I updated!!